examples/scene-example.ts

```diff
@@ -0,0 +1,378 @@
+
+        /* Create and configure forward pass. */
+
+        this._forwardPass = new ForwardSceneRenderPass(context);
+        this._forwardPass.initialize();
+
+        this._forwardPass.camera = this._camera;
+        this._forwardPass.target = this._framebuffer;
+
+        /* Create scene. */
+
+        this.generateScene();
+        this._forwardPass.scene = this._scene;
+
+        return true;
+    }
+
+    /**
+     * Uninitializes Buffers, Textures, and Program.
+     */
+    protected onUninitialize(): void {
+        super.uninitialize();
+
+        // TODO: make sure that all meshes and programs inside of the scene get cleaned
+
+        // this._mesh.uninitialize();
+        // this._meshProgram.uninitialize();
+    }
+
+    /**
+     * This is invoked in order to check if rendering of a frame is required by means of implementation specific
+     * evaluation (e.g., lazy non continuous rendering). Regardless of the return value a new frame (preparation,
+     * frame, swap) might be invoked anyway, e.g., when update is forced or canvas or context properties have
+     * changed or the renderer was invalidated @see{@link invalidate}.
+     * Updates the navigaten and the AntiAliasingKernel.
+     * @returns whether to redraw
+     */
+    protected onUpdate(): boolean {
+        const gl = this.context.gl;
+        // Resize
+        if (this._altered.frameSize) {
+            this._camera.viewport = [this._frameSize[0], this._frameSize[1]];
+            gl.viewport(0, 0, this._frameSize[0], this._frameSize[1]);
+        }
+        if (this._altered.canvasSize) {
+            this._camera.aspect = this._canvasSize[0] / this._canvasSize[1];
+        }
+
+        // Update clear color
+        if (this._altered.clearColor) {
+            const c = this._clearColor;
+            gl.clearColor(c[0], c[1], c[2], c[3]);
+        }
+
+        this._navigation.update();
+        this._forwardPass.update();
+
+        return this._altered.any || this._camera.altered;
+    }
+
+    /**
+     * This is invoked in order to prepare rendering of one or more frames, regarding multi-frame rendering and
+     * camera-updates.
+     */
+    protected onPrepare(): void {
+        this._forwardPass.prepare();
+
+        this._altered.reset();
+        this._camera.altered = false;
+    }
+
+    /**
+     * @todo comment
+     * @param frameNumber - for intermediate frames in accumulation rendering.
+     */
+    protected onFrame(frameNumber: number): void {
+        const gl = this._context.gl;
+
+        this._camera.viewport = [this._frameSize[0], this._frameSize[1]];
+
+        gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+        gl.enable(gl.CULL_FACE);
+        gl.cullFace(gl.BACK);
+        gl.enable(gl.DEPTH_TEST);
+
+        this._forwardPass.frame();
+
+        gl.cullFace(gl.BACK);
+        gl.disable(gl.CULL_FACE);
+    }
+
+    /**
+     * @todo comment ...
+     */
+    protected onSwap(): void {
```

Comment:
Just for comment: The expected functionality is to ensure the contents of a frame to be on the OpenGL backbuffer. Is this example renders into the backbuffer, nothing has to be done. The swap of front buffer and back buffer is scheduled after the invocation of this function.